src/AvTranscoder/encoder/AudioEncoder.cpp

```diff
@@ -112,24 +112,21 @@ bool AudioEncoder::encodeFrame(const Frame& sourceFrame, CodedData& codedFrame)
 #if LIBAVCODEC_VERSION_MAJOR > 53
     int gotPacket = 0;
     int ret = avcodec_encode_audio2(&avCodecContext, &packet, &sourceFrame.getAVFrame(), &gotPacket);
-    if(ret != 0 && gotPacket == 0)
+    if(ret != 0)
     {
         throw std::runtime_error("Encode audio frame error: avcodec encode audio frame - " +
                                  getDescriptionFromErrorCode(ret));
     }
+    return ret == 0 && gotPacket == 1;
```

Comment:
The condition _ret == 0_ is not necessary here, because if it's false, you throw a runtime error (just before) !